src/domain/auth/login.use-case.ts

```diff
@@ -16,6 +16,32 @@ interface LoginUseCaseResponse {
   token: string;
 }
 
+export class LoginUseCase {
+  constructor(private readonly repository = AppDataSource.getRepository(User)) {}
+
+  async execute(input: LoginInput): Promise<LoginUseCaseResponse> {
+    const databaseUser = await this.repository.findOneBy({ email: input.email });
+
+    const isSamePassword = await bcrypt.compare(input.password, databaseUser.password);
+
+    if (!isSamePassword) {
+      throw new CustomError('Invalid credentials', 401);
+    }
```

Comment:
existe a possibilidade de nao encontrar o usuário, certo?
deveria jogar um erro pq ai significa que o email esta errado
```suggestion
    const user = await this.repository.findOneBy({ email: input.email });

    const isSamePassword = await bcrypt.compare(input.password, user?.password);

    if (!user || !isSamePassword) {
      throw new CustomError('Invalid credentials', 401);
    }
```